src/main/java/reposense/system/CommandRunner.java

```diff
@@ -108,6 +112,24 @@ public static String getCommitHashBeforeDate(String root, String branchName, Dat
         return runCommand(rootPath, revListCommand);
     }
 
+    /**
+     * Removes the author for the {@code commitHash} in the {@code branch} of the repo.
+     */
+    public static void removeCommitAuthor(String root, String branch, String commitHash) {
+        Path rootPath = Paths.get(root);
+        // create temp branch with the given commitHash as the HEAD
+        runCommand(rootPath, String.format(CHECKOUT_COMMIT_IN_TEMP_BRANCH_COMMAND_FORMAT, commitHash));
+
+        // remove the author for the given commit
+        runCommand(rootPath, REMOVE_COMMIT_AUTHOR_COMMAND);
+
+        // rebase temp branch onto the given branch, replacing the given commit's author on that branch
+        runCommand(rootPath, String.format(REBASE_ONTO_MASTER_FROM_TEMP_BRANCH_COMMAND_FORMAT, branch));
```

Comment:
shouldn't it be rebase master onto the given branch?

Reply:
nope, we are doing the editing on the given branch 1st, then rebasing those changes into master branch, as the master branch that we edit is temporary, and won't affect the original.